numpy/random/_generator.pyi

```diff
@@ -0,0 +1,245 @@
+        size: Union[None, _ShapeLike] = ...,
+    ) -> Union[float, ndarray[Any, dtype[float64]]]: ...
+    def triangular(
+        self,
+        left: _ArrayLikeFloat_co,
+        mode: _ArrayLikeFloat_co,
+        right: _ArrayLikeFloat_co,
+        size: Union[None, _ShapeLike] = ...,
+    ) -> Union[float, ndarray[Any, dtype[float64]]]: ...
+    # Complicated, discrete distributions:
+    def binomial(
+        self, n: _ArrayLikeInt_co, p: _ArrayLikeFloat_co, size: Union[None, _ShapeLike] = ...
+    ) -> Union[int, ndarray[Any, dtype[int64]]]: ...
+    def negative_binomial(
+        self, n: _ArrayLikeFloat_co, p: _ArrayLikeFloat_co, size: Union[None, _ShapeLike] = ...
+    ) -> Union[int, ndarray[Any, dtype[int64]]]: ...
+    def poisson(
+        self, lam: _ArrayLikeFloat_co = ..., size: Union[None, _ShapeLike] = ...
+    ) -> Union[int, ndarray[Any, dtype[int64]]]: ...
+    def zipf(
+        self, a: _ArrayLikeFloat_co, size: Union[None, _ShapeLike] = ...
+    ) -> Union[int, ndarray[Any, dtype[int64]]]: ...
+    def geometric(
+        self, p: _ArrayLikeFloat_co, size: Union[None, _ShapeLike] = ...
+    ) -> Union[int, ndarray[Any, dtype[int64]]]: ...
+    def hypergeometric(
+        self,
+        ngood: _ArrayLikeInt_co,
+        nbad: _ArrayLikeInt_co,
+        nsample: _ArrayLikeInt_co,
+        size: Union[None, _ShapeLike] = ...,
+    ) -> Union[int, ndarray[Any, dtype[int64]]]: ...
+    def logseries(
+        self, p: _ArrayLikeFloat_co, size: Union[None, _ShapeLike] = ...
+    ) -> Union[int, ndarray[Any, dtype[int64]]]: ...
+    # Multivariate distributions:
+    # TODO: Really need  1-d array like floating and 2-d array-like floating. Using Sequence[float] ??
+    def multivariate_normal(
+        self,
+        mean: Sequence[float],
+        cov: Sequence[Sequence[float]],
+        size: Union[None, _ShapeLike] = ...,
+        check_valid: Literal["warn", "raise", "ignore"] = ...,
+        tol: float = ...,
+        *,
+        method: Literal["svd", "eigh", "cholesky"] = ...
+    ): ...
+    # TODO: Need 1-d array like floating. Using Sequence[float] ??
+    def multinomial(
+        self, n: _ArrayLikeInt_co, pvals: Sequence[float], size: Union[None, _ShapeLike] = ...
+    ): ...
+    # TODO: Need 1-d array like integers. Using Sequence[int] ??
+    def multivariate_hypergeometric(
+        self,
+        colors: Sequence[int],
+        nsample: int,
+        size: Union[None, _ShapeLike] = ...,
+        method: Literal["marginals", "count"] = ...,
+    ): ...
+    # TODO: Need 1-d array like floating. Using Sequence[float] ??
+    def dirichlet(
+        self, alpha: Sequence[float], size: Union[None, _ShapeLike] = ...
+    ) -> ndarray[Any, dtype[float64]]: ...
+    def permuted(
+        self, x: ArrayLike, *, axis: Optional[int] = ..., out: Optional[ndarray[Any, Any]] = ...
+    ) -> ndarray[Any, Any]: ...
+    def shuffle(self, x: ArrayLike, axis: int = ...) -> Sequence[Any]: ...
```

Comment:
```suggestion
    def shuffle(
        self, x: Union[MutableSequence[Any], ndarray[Any, Any]], axis: int = ...
    ) -> None: ...
```
Two things:
* `shuffle` performs inplace operations.
* It's a bit more restrictive in its accepted input types, demanding a mutable sequence or actual `ndarray`.